### DISCUSSION TITLE
Dashboard: Allow editing json of broken dashboards

### DISCUSSION THREAD

**Discussion by sylr** [grafana/grafana#38125] [Dashboards, Panels & Visualizations]
**What happened**:

I Made a circular variable dependency in a dashboard, did not get any error because while doing it all variables where initialized so it worked.

When I opened up the dashboard with no context the dashboard refused to load, got a `<spinner gif> Services` and a `cancel loading dashboard` button.

After a while I clicked the cancel button and got this error:

```
Templating init failed Couldn't find variable with id:nodelifecycle
```

At this point we are left with no means to edit the dashboard to fix the mistake and people with no database access or SQL knowledge are left with no other choice than redo the dashboard from scratch.

**What you expected to happen**:

Appending `&editview=dashboard_json` to the dashboard url would allow me to edit the dashboard JSON so that I can remove the circular dependency.

**How to reproduce it (as minimally and precisely as possible)**:

Create a dashboard with a variable circular dependency, save it, kill your browser, open the dashboard again, get the spinner.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.0.3
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by jacksongoode** [comment]
I think this has a similar usage to this question #47836 (though my question is less exposed). I'm wondering if you have had any ability to edit the JSON model within the browser to preview the changes you made prior to posting?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
